fizz-buzz-streams/src/test/java/com/pubsap/eng/streams/FizzBuzzAllTopologySpec.java

```diff
@@ -0,0 +1,192 @@
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.streams.*;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import io.confluent.kafka.schemaregistry.client.MockSchemaRegistryClient;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.*;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class FizzBuzzAllTopologySpec {
+
+
+    private TopologyTestDriver testDriver;
+    private TestInputTopic<InputKey, Input> inputTopic;
+    private TestOutputTopic<OutputKey, Output> outputTopic;
+
+    @Before
+    public void setTopologyTestDriver() throws IOException, RestClientException {
+
+        final String srConfigKey = "schema.registry.url";
+
+        final Config config = ConfigFactory.load();
+        Properties properties = new Properties();
+
+        properties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, config.getString("bootstrap.servers"));
+        properties.put(StreamsConfig.APPLICATION_ID_CONFIG, config.getString("application.id"));
+        properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
+        properties.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
+
+
+        String inputTopicName = config.getString("input.topic.name");
+        String outputTopicName = config.getString("output.topic.name");
+
+        MockSchemaRegistryClient mockedRegistry = new MockSchemaRegistryClient();
+
+
+        SpecificAvroSerde<Item> itemSerde = new SpecificAvroSerde<>(mockedRegistry);
+
+        SpecificAvroSerde<InputKey> inputKeySerde = new SpecificAvroSerde<>(mockedRegistry);
+        SpecificAvroSerde<Input> inputSerde = new SpecificAvroSerde<>(mockedRegistry);
+
+        SpecificAvroSerde<OutputKey> outputKeySerde = new SpecificAvroSerde<>(mockedRegistry);
+        SpecificAvroSerde<Output> outputSerde = new SpecificAvroSerde<>(mockedRegistry);
+
+        itemSerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), false);
+        inputSerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), false);
+        outputSerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), false);
+        inputKeySerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), true);
+        outputKeySerde.configure(Collections.singletonMap(srConfigKey, config.getString(srConfigKey)), true);
+
+        mockedRegistry.register(inputTopic + "-value", Input.SCHEMA$);
+        mockedRegistry.register(outputTopic + "-value", Output.SCHEMA$);
+
+        TimeWindows windows = TimeWindows
+
+                .of(Duration.ofSeconds(20))
```

Comment:
Latest version creates the window from the `Config` object. This can be applied here.